SpatialGDK/Source/SpatialGDK/Private/Utils/SpatialDebugger.cpp

```diff
@@ -282,7 +284,65 @@ void ASpatialDebugger::OnEntityAdded(const Worker_EntityId EntityId)
 		if (Actor->IsA<APlayerController>())
 		{
 			LocalPlayerController = Cast<APlayerController>(Actor);
+
+			if (ConfigUIWidget == nullptr)
+			{
+				FSoftClassPath ConfigUIClassRef(DEFAULT_CONFIG_UI_WIDGET_CLASS);
+				UClass* ConfigUIClass = ConfigUIClassRef.TryLoadClass<UUserWidget>();
+				if (ConfigUIClass)
+				{
+					ConfigUIWidget = CreateWidget<USpatialDebuggerConfigUI>(LocalPlayerController.Get(), ConfigUIClass);
+					ConfigUIWidget->SetSpatialDebugger(this);
+				}
+			}
+			LocalPlayerController->InputComponent->BindKey(ConfigUIKey, IE_Pressed, this, &ASpatialDebugger::OnToggleConfigUI);
+		}
+	}
+}
+
+void ASpatialDebugger::OnToggleConfigUI()
+{
+	if (ConfigUIWidget == nullptr)
+	{
+		return;
+	}
+
+	if (!ConfigUIWidget->IsInViewport())
+	{
+		ConfigUIWidget->AddToViewport();
+
+		// TODO This is clobbering whatever input mode the player controller had set before we enabled our UI.
```

Comment:
I think we should store the old one in a var 👍

Reply:
I looked into it further, and the input mode doesn't get stored anywhere, meaning that we can't retrieve it before changing it. Applying an input mode also touches a bunch of state, and storing that state before applying the input mode seems difficult if not impossible.
So it doesn't look like showing this UI can be done completely invisibly to user code.
My best alternate idea would be a callback for user code to hook into and restore state as it needs to, but that'd require some additional code to make it easy for users to subscribe to that callback when the spatial debugger is created.